fvm/src/call_manager/default.rs

```diff
@@ -955,3 +1040,53 @@ impl EventsAccumulator {
         })
     }
 }
+
+impl Entrypoint {
+    fn method_num(&self) -> MethodNum {
+        match self {
+            Entrypoint::Invoke(num) => *num,
+            Entrypoint::Upgrade(_) => fvm_shared::METHOD_UPGRADE,
+        }
+    }
+
+    fn func_name(&self) -> &'static str {
+        match self {
+            Entrypoint::Invoke(_) => "invoke",
+            Entrypoint::Upgrade(_) => "upgrade",
+        }
+    }
+}
+
+// EntrypointParams is a helper struct to init the registry with the entrypoint specific
+// parameters and then forward them to wasmtime
+struct EntrypointParams {
+    entrypoint: Entrypoint,
+    params: Vec<wasmtime::Val>,
+}
+
+impl EntrypointParams {
+    fn new(entrypoint: Entrypoint) -> Self {
+        Self {
+            entrypoint,
+            params: Vec::new(),
+        }
+    }
+
+    fn maybe_put_registry(&mut self, br: &mut BlockRegistry) -> Result<()> {
```

Comment:
So, if we go the trait route, I'd define it as:

```rust
trait Entrypoint {
    fn into_params(self, br: &mut BlockRegistry) -> Result<EntryPointParams>;
}
```

This means there's no concerns about `maybe_put_registry` being called multiple times (e.g.).